packages/integration-sdk-cli/src/bocchi/templates/top-level/src/client.ts.hbs

```diff
@@ -0,0 +1,115 @@
+import { BaseAPIClient } from '@jupiterone/integration-sdk-http-client';
+import { IntegrationConfig } from './config';
+import { 
+{{#each template.steps}}
+  {{pascalCase entity.name}}, 
+{{/each}}
+} from './steps/types';
+import { Entity, IntegrationLogger } from '@jupiterone/integration-sdk-core';
+
+export type ResourceIteratee<T> = (each: T) => Promise<void> | void;
+
+export class APIClient extends BaseAPIClient {
+  private authHeaders: Record<string, string>;
+
+  constructor(readonly config: IntegrationConfig, logger: IntegrationLogger) {
+    super({
+      baseUrl: '{{template.baseUrl}}',
+      logger,
+      {{#if template.tokenBucket}}
+      tokenBucket: {
+        maximumCapacity: {{template.tokenBucket.maximumCapacity}},
+        refillRate: {{template.tokenBucket.refillRate}}
+      }
+      {{/if}}
+    });
+  }
+
+  {{#if (isNotEndpointAuth template.authentication.strategy)}}
+  protected getAuthorizationHeaders(): Record<string, string> {
+    return { {{#each (sanitizeAuthObject template.authentication.authHeaders)}}'{{@key}}': `{{this}}`,{{/each}} };
+  }  
+  {{else}}
+  protected async getAuthorizationHeaders(): Promise<Record<string, string>> {
+    const response = await (await this.retryableRequest(
+      `${this.baseUrl}{{template.authentication.params.path}}`,
+      {
+        headers: { {{#each (sanitizeAuthObject template.authentication.params.headers)}}'{{@key}}': `{{this}}`,{{/each}} },
+        method: '{{sanitizeHttpMethod template.authentication.params.method}}',
```

Comment:
@RonaldEAM `if it hasn't been called before` but this is what I want, no? Like if I don't have `authorize: false`, it will just call getAuthorizationHeaders, set them and then never call it again. That's what I want, no? I've been running the resulting graph project without `authorize: false` and it has been working.

Reply:
OH I understand. We need `authorize: false` for this function only, got it. I'll update that.